modules/javafx.controls/src/main/java/javafx/scene/control/skin/ToolBarSkin.java

```diff
@@ -566,8 +568,18 @@ private void initialize() {
         box.getStyleClass().add("container");
         box.getChildren().addAll(getSkinnable().getItems());
         // The overflowBox must have the same style classes, otherwise the overflow items may get wrong values.
-        overflowBox.setId(box.getId());
-        overflowBox.getStyleClass().addAll(box.getStyleClass());
+        overflowBox.idProperty().bind(box.idProperty());
+        overflowBox.getStyleClass().setAll(box.getStyleClass());
+        box.getStyleClass().addListener((ListChangeListener<? super String>) change -> overflowBox.getStyleClass().setAll(change.getList()));
```

Comment:
I think what you need here (and below) is `Bindings.bindContent(List, ObservableList)`

Reply:
Good suggestion